### PR TITLE
Fix researchgate

### PR DESCRIPTION
**Pull request by GuyAglionby**
(Supersedes #1828 -- they made further changes to the site) Update to new URL for RIS + fix detection code for searching.

Possible URL patterns for searching:

* https://www.researchgate.net/search?q=zotero
* https://www.researchgate.net/search/publications?q=zotero
* https://www.researchgate.net/search (URL not updated when performing a search)

Thanks!

### CODE REVIEW

File: ResearchGate.js [hunk 1]
```diff
@@ -37,36 +37,41 @@
 
 
 // attr()/text() v2
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null}
+function attr(docOrElem, selector, attr, index) {
+	var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector); return elem ? elem.getAttribute(attr) : null;
+} function text(docOrElem, selector, index) {
+	var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector); return elem ? elem.textContent : null;
+}
```
Comment (dstillman) [inline ResearchGate.js]: Undo. (We'll be getting rid of these lines soon, but for now the linter changes this incorrectly.)

Comment (dstillman) [reply ResearchGate.js]: Thanks — can you squash that into the previous commit just to avoid the extra churn?

Comment (GuyAglionby) [reply ResearchGate.js]: Done! thanks

**Comment by adam3smith** [comment]
Thanks! For future reference, merging in spite of tests failures as these are only due to minified `attr/text` code.

**Pull request merged**